guacamole/src/main/java/org/apache/guacamole/rest/directory/DirectoryResource.java

```diff
@@ -590,7 +591,56 @@ public void executeOperation(boolean atomic, Directory<InternalType> directory)
 
                     }
 
-                    // Append each identifier to the list, to be removed atomically
+                    else if (op == APIPatch.Operation.replace) {
+
+                        // The identifier of the object to be replaced
+                        String identifier = path.substring(1);
+
+                        InternalType original = null;
+
+                        try {
+
+                            // Fetch the object to be updated
+                            original = directory.get(identifier);
```

Comment:
`get()` has its own success/failure event type that should be dispatched separately from `update()`. See:

https://github.com/apache/guacamole-client/blob/05c675bb215111643fc1213f661b07867a98aeb0/guacamole/src/main/java/org/apache/guacamole/rest/directory/DirectoryResource.java#L747-L757

Reply:
Ok, I've updated it. I think this is the right logic now - if an attempt is made to replace a non-existant object, it will now log both a GET and UPDATE failure event.

Reply:
If a user attempts to update a single connection via the non-batch REST API, but that connection does not exist, they'll currently get a `GET` failure event but no `UPDATE` failure event (the attempt to update doesn't occur because the retrieval failed). Do you think these changes should match that behavior, or that the REST API changes that are part of GUACAMOLE-1224 should be modified to ensure both are fired?

Reply:
Hmm, it seems to me that a failed attempt to update a record (even if it's because it doesn't exist), should trigger an UPDATE failure event. What do you think?

Reply:
Yeah, I agree. There'd otherwise be no context for failures in audit logs.